src/test/shell/bazel/target_compatible_with_test.sh

```diff
@@ -0,0 +1,378 @@
+  expect_log 'Target //:pass_on_foo1 was skipped'
+  expect_log 'Target //:pass_on_foo1_bar2 was skipped'
+  expect_not_log 'Target //:fail_on_foo2 failed to build'
+  expect_not_log 'Target //:pass_on_foo1 failed to build'
+  expect_not_log 'Target //:pass_on_foo1_bar2 failed to build'
+  expect_not_log 'Target //:fail_on_foo2 up-to-date'
+  expect_not_log 'Target //:pass_on_foo1 up-to-date'
+  expect_not_log 'Target //:pass_on_foo1_bar2 up-to-date'
+}
+
+function test_console_log_for_tests() {
+  cd target_skipping || fail "couldn't cd into workspace"
+
+  # Get repeatable results from this test.
+  bazel clean --expunge
+
+  if ! bazel test --host_platform=@//:foo1_bar1_platform \
+    --platforms=@//:foo1_bar1_platform ... &> "${TEST_log}"; then
+    cat "${TEST_log}"
+    fail "Bazel failed unexpectedly."
+  fi
+  expect_log 'Executed 1 out of 3 tests: 1 test passes and 2 were skipped'
+  expect_log '^//:pass_on_foo1  *  PASSED in'
+  expect_log '^//:fail_on_foo2  *  SKIPPED$'
+  expect_log '^//:pass_on_foo1_bar2  *  SKIPPED$'
+
+  if bazel test --host_platform=@//:foo2_bar1_platform \
+    --platforms=@//:foo2_bar1_platform ... &> "${TEST_log}"; then
+    cat "${TEST_log}"
+    fail "Bazel passed unexpectedly."
+  fi
+  expect_log 'Executed 1 out of 3 tests: 1 fails locally and 2 were skipped.'
+  expect_log '^//:pass_on_foo1  *  SKIPPED$'
+  expect_log '^//:fail_on_foo2  *  FAILED in'
+  expect_log '^//:pass_on_foo1_bar2  *  SKIPPED$'
+
+  # Use :all for this one to validate similar behaviour.
+  if ! bazel test --host_platform=@//:foo1_bar2_platform \
+    --platforms=@//:foo1_bar2_platform :all &> "${TEST_log}"; then
+    cat "${TEST_log}"
+    fail "Bazel failed unexpectedly."
+  fi
+  expect_log 'Executed 2 out of 3 tests: 2 tests pass and 1 was skipped'
+  expect_log '^//:pass_on_foo1  *  PASSED in'
+  expect_log '^//:fail_on_foo2  *  SKIPPED$'
+  expect_log '^//:pass_on_foo1_bar2  *  PASSED in'
+}
+
+function test_failure_on_incompatible_top_level_target() {
+  cd target_skipping || fail "couldn't cd into workspace"
+
+  if bazel test --show_result=10 --host_platform=@//:foo1_bar1_platform \
+    --platforms=@//:foo1_bar1_platform //:pass_on_foo1_bar2 \
+    --build_event_text_file="${TEST_log}".build.json \
+    &> "${TEST_log}"; then
+    fail "Bazel passed unexpectedly."
+  fi
+
+  expect_log 'ERROR: Target //:pass_on_foo1_bar2 is incompatible and cannot be built'
+  expect_log '^FAILED: Build did NOT complete successfully'
+
+  # Now look at the build event log.
+  mv "${TEST_log}".build.json "${TEST_log}"
+
+  expect_log '^    name: "PARSING_FAILURE"$'
+  expect_log 'Target //:pass_on_foo1_bar2 is incompatible and cannot be built.'
+
+  # Run an additional (passing) test and make sure we still fail the build.
+  if bazel test --show_result=10 --host_platform=@//:foo1_bar1_platform \
+    --platforms=@//:foo1_bar1_platform //:pass_on_foo1_bar2 //:pass_on_foo1 \
+    --keep_going \
```

Comment:
Naive question: is this the part that tests the difference between `-keep_going` and non-`keep_going` failures?

Reply:
It is. Let me add some comments.